locust/web.py

```diff
@@ -45,8 +47,23 @@ def __init__(self, environment):
         self.app = app
         app.debug = True
         app.root_path = os.path.dirname(os.path.abspath(__file__))
+        app.config["BASIC_AUTH_ENABLED"] = False
+        self.auth = BasicAuth()
+
+        def auth_required_if_enabled(view_func):
```

Comment:
Looking at the [documentation](https://flask-basicauth.readthedocs.io/en/latest/) for Flask-BasicAuth, there seem to be an option called `BASIC_AUTH_FORCE`. Couldn't we use that and not have to decorate every endpoint?

Reply:
Hmm, but the `auth_required_if_enabled` isn't exposed in the API as far as I can tell?

Reply:
You're right !  I have made relevant changes now which exposes the decorator to be added to custom routes.